programs/pkey/pk_verify.c

```diff
@@ -57,11 +57,22 @@ int main( int argc, char *argv[] )
     FILE *f;
     int ret = 1;
     size_t i;
+#if defined(MBEDTLS_PLATFORM_C)
+    mbedtls_platform_context platform_ctx;
+#endif
     mbedtls_pk_context pk;
     unsigned char hash[32];
     unsigned char buf[MBEDTLS_MPI_MAX_SIZE];
     char filename[512];
 
+#if defined(MBEDTLS_PLATFORM_C)
+    if( ( ret = mbedtls_platform_setup( &platform_ctx ) ) != 0 )
+    {
+        mbedtls_printf( " failed\n  ! mbedtls_platform_setup returned %d\n", ret );
+        goto exit;
```

Comment:
This should be changed to `return ret`. If you go to exit here, you will free uninitialized resources, which might lead to undefined behavior, depending on implementation.